frontend/src/pages/index.js

```diff
@@ -1,49 +1,82 @@
-import React, { Component } from 'react'
+import React from 'react'
+import { Component } from 'react'
 import Router from 'next/router'
-import { Button, Container, Input, Row, Col } from 'reactstrap'
-import { validateKey } from '../utils/api'
-import ReactLoading from 'react-loading'
+import { loginUser, loginGoogleUser } from '../utils/api'
 import Nav from '../components/nav'
 import Head from '../components/head'
+import { GoogleLogin } from 'react-google-login'
+import {
+  Container,
+  Form,
+  Button,
+  FormGroup,
+  Label,
+  Input,
+  Card,
+  CardBody,
+  CardTitle,
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter
+} from 'reactstrap'
+import cookie from 'js-cookie'
 
 class LoginPage extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      currentKey: '',
-      loading: false
+      email: '',
+      password: '',
+      errorMessage: '',
+      showInvalidRequestModal: false
     }
   }
 
-  handleSubmit = async () => {
-    this.setState({
-      loading: true
-    })
-    const { success, result } = await validateKey(this.state.currentKey)
-    if (success) {
-      localStorage.setItem('interviewerKey', this.state.currentKey)
-      localStorage.setItem('interviewerName', result.name)
-      this.setState({
-        loading: false
+  setCookie = (key, value) => {
+    if (process.browser) {
+      cookie.set(key, value, {
+        expires: 1,
+        path: '/'
       })
-      Router.push('/dashboard')
+    }
+  }
+
+  handleGoogle = async e => {
+    const result = await loginGoogleUser(e.tokenId)
+    const resp = await result.json()
+    if (resp.status !== 200) {
+      this.setState({ errorMessage: resp.message, showInvalidRequestModal: true })
+      console.log(resp.message)
     } else {
-      this.setState({
-        loading: false
-      })
-      alert("Couldn't log in. Is your key correct?")
+      this.setCookie('token', e.tokenId)
+      this.setCookie('google', true)
+      localStorage.setItem('interviewerKey', 'ohno') // TODO: Create switch statements for roles
```

Comment:
Also when putting `TODOs` in comments, please create GitHub issues and link them. Even for things you might not get to its very helpful for future development.